protocol/chainlib/jsonRPC.go

```diff
@@ -423,6 +424,10 @@ func (cp *JrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 	// the error check here would only wrap errors not from the rpc
 	if err != nil {
 		utils.LavaFormatDebug("received an error from SendNodeMsg", utils.Attribute{Key: "GUID", Value: ctx}, utils.Attribute{Key: "error", Value: err})
+		if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
```

Comment:
string contains is not the way to check if there was a context deadline you have connectCtx.Error():
ctx.Err() == context.DeadlineExceeded